test/envoye2e/env/setup.go

```diff
@@ -432,6 +445,69 @@ func (s *TestSetup) VerifyStats(expectedStats map[string]int, port uint16) {
 	s.t.Errorf("failed to find expected stats: %v", err)
 }
 
+// VerifyPrometheusStats verifies prometheus stats.
+func (s *TestSetup) VerifyPrometheusStats(expectedStats map[string]int, port uint16) {
+	s.t.Helper()
+
+	check := func(respBody string) error {
+		for eStatsName, eStatsValue := range expectedStats {
+			var parser expfmt.TextParser
+			reader := strings.NewReader(respBody)
+			mapMetric, err := parser.TextToMetricFamilies(reader)
+			if err != nil {
+				return err
+			}
+			aStats, ok := mapMetric[eStatsName]
+			if !ok {
+				return fmt.Errorf("failed to find expected stat %s", eStatsName)
+			}
+			var aStatsValue float64
+			switch aStats.GetType() {
+			case dto.MetricType_COUNTER:
+				log.Printf("found counter")
+				if len(aStats.GetMetric()) != 1 {
+					return fmt.Errorf("expected one value for counter")
+				}
+				aStatsValue = (*(aStats.GetMetric())[0]).GetCounter().GetValue()
```

Comment:
You don't need this pointer dance. `aStats.GetMetric()[0].GetCounter()` works.

Reply:
Thanks..Fixed